lib/ibmdb.js

```diff
@@ -536,116 +536,206 @@ IBMDB.prototype.updateOrCreate = IBMDB.prototype.save =
  * or create a new instance
  *
  * @param {string} model The model name
+ * @param {Object} where clause
  * @param {Object} data The model instance data
+ * @param {Object} options for this call
  * @param {Function} [callback] The callback function
  */
-IBMDB.prototype.replaceOrCreate = IBMDB.prototype.replace =
-  function(model, data, options, callback) {
-    debug('IBMDB.prototype.replaceOrCreate (enter): model=%j, data=%j, ' +
-          'options=%j\n', model, data, options);
-    var self = this;
-    var idName = self.idName(model);
-    var stmt;
-    var tableName = self.tableEscaped(model);
-    var meta = {};
-
-    function executeWithConnection(connection, cb) {
-      // Execution for replaceOrCreate requires running 3
-      // separate SQL statements. The last depends on the
-      // result of the first couple.
-      var where = {};
-      where[idName] = data[idName];
-
-      var countStmt = new ParameterizedSQL('SELECT COUNT(*) AS CNT FROM ');
-      countStmt.merge(tableName);
-      countStmt.merge(self.buildWhere(model, where));
-      countStmt.noResults = false;
-
-      connection.query(countStmt, function(err, countData) {
-        debug('IBMDB.prototype.replaceOrCreate (data): err=%j, countData=%j\n',
-              err, countData);
-        if (err) return cb(err);
+IBMDB.prototype._replace = function(model, where, data, options, callback) {
+  debug('IBMDB.prototype._replace (enter): model=%j, data=%j, ' +
+        'options=%j\n', model, data, options);
+  var self = this;
+  var stmt;
+  var tableName = self.tableEscaped(model);
+  var meta = {};
+
+  function executeWithConnection(connection, cb) {
+    // Execution for _replace requires running 3
+    // separate SQL statements. The last depends on the
+    // result of the first couple.
+
+    var countStmt = new ParameterizedSQL('SELECT COUNT(*) AS CNT FROM ');
+    countStmt.merge(tableName);
+    countStmt.merge(self.buildWhere(model, where));
+    countStmt.noResults = false;
+
+    connection.query(countStmt, function(err, countData) {
+      debug('IBMDB.prototype._replace stmt: %j data: %j err: %j\n',
+            countStmt, countData, err);
+      if (err) return cb(err);
+
+      if (countData[0]['CNT'] > 0) {
+        // remove existing to replace with a new insert
+        stmt = self.buildUpdate(model, where, data);
```

Comment:
_replace should do the same .. Will fix now, this showed up in the DB2 tests when ran anyway.